### ISSUE TITLE
dict-keys-not-iterating false positive when used with `in`

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#2186]
### Steps to reproduce
Run pylint over the following example:
```
a = {}
2 in a.keys()
``` 

### Current behavior
We're currently emitting `dict-keys-not-iterating`

### Expected behavior

We shouldn't emit it as the containment is an iteration context.

### pylint --version output
1.9.2, didn't try with `master`.
